sql/src/main/java/io/crate/operation/aggregation/impl/PercentileAggregation.java

```diff
@@ -0,0 +1,157 @@
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.aggregations.metrics.percentiles.tdigest.TDigestState;
+
+import java.io.IOException;
+
+
+public class PercentileAggregation extends AggregationFunction<TDigestState, Double> {
+
+    public static final String NAME = "percentile";
+    private static final int COMPRESSION = 100;
+
+    private FunctionInfo info;
+    private double fraction = 0.0;
+
+    static {
+        DataTypes.register(TDigestStateType.ID, TDigestStateType.INSTANCE);
+    }
+
+    public static void register(AggregationImplModule mod) {
+        for (final DataType dataType : DataTypes.NUMERIC_PRIMITIVE_TYPES) {
+            mod.register(new PercentileAggregation(new FunctionInfo(new FunctionIdent(NAME,
+                    ImmutableList.of(dataType, DataTypes.DOUBLE)),
+                    DataTypes.DOUBLE, FunctionInfo.Type.AGGREGATE)));
+        }
+    }
+
+    PercentileAggregation(FunctionInfo info) {
+        this.info = info;
+    }
+
+    public static class TDigestStateType extends DataType<TDigestState> implements Streamer<TDigestState>, DataTypeFactory {
+
+        private static final int ID = 5120;
+        private static final TDigestStateType INSTANCE = new TDigestStateType();
+
+
+        @Override
+        public DataType<?> create() {
+            return INSTANCE;
+        }
+
+        @Override
+        public TDigestState readValueFrom(StreamInput in) throws IOException {
+            return TDigestState.read(in);
+        }
+
+        @Override
+        public void writeValueTo(StreamOutput out, Object v) throws IOException {
+            TDigestState.write((TDigestState) v, out);
+        }
+
+        @Override
+        public int id() {
+            return ID;
+        }
+
+        @Override
+        public String getName() {
+            return "percentile_state";
+        }
+
+        @Override
+        public Streamer<?> streamer() {
+            return this;
+        }
+
+        @Override
+        public TDigestState value(Object value) throws IllegalArgumentException, ClassCastException {
+            return (TDigestState) value;
+        }
+
+        @Override
+        public int compareValueTo(TDigestState val1, TDigestState val2) {
+            return 0;
+        }
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    public TDigestState newState(RamAccountingContext ramAccountingContext) {
+        ramAccountingContext.addBytes(DataTypes.DOUBLE.fixedSize()); // overhead for the HashSet (map ref 8 + 28 for fields inside the map)
+        return new TDigestState(COMPRESSION);
+    }
+
+    @Override
+    public TDigestState iterate(RamAccountingContext ramAccountingContext, TDigestState state, Input... args) throws CircuitBreakingException {
+        state.add(DataTypes.DOUBLE.value(args[0].value()));
+        fraction = DataTypes.DOUBLE.value(args[1].value());
```

Comment:
fraction must become part of the state object and must not be part of the aggregation function itself. 
E.g. if you have something like this
```
  CollectPhase
    (iterate )
         |
     send state to other node
    (TDigestState)
         |
   MergePhase
   ( reduce)
       |
       |
    send state to other node
    (TDigestState)
      |
      |
   terminatePartial
   (has no fraction here,
   because iterate isn't called here)

```